crates/viewer/re_renderer/src/renderer/video.rs

```diff
@@ -0,0 +1,544 @@
+use re_video::VideoLoadError;
+
+use crate::resource_managers::GpuTexture2D;
+use crate::wgpu_resources::GpuTexturePool;
+use crate::wgpu_resources::TextureDesc;
+use crate::RenderContext;
+
+use re_video::TimeMs;
+
+pub struct Video {
+    decoder: decoder::VideoDecoder,
+}
+
+impl Video {
+    pub fn load(
+        render_context: &RenderContext,
+        media_type: Option<&str>,
+        data: &[u8],
+    ) -> Result<Self, VideoError> {
+        let data = match media_type {
+            Some("video/mp4") => re_video::load_mp4(data)?,
+            Some(media_type) => {
+                return Err(VideoError::Load(VideoLoadError::UnsupportedMediaType(
+                    media_type.to_owned(),
+                )))
+            }
+            None => return Err(VideoError::Load(VideoLoadError::UnknownMediaType)),
+        };
+        let decoder = VideoDecoder::new(render_context, data).ok_or_else(|| VideoError::Init)?;
+
+        Ok(Self { decoder })
+    }
+
+    pub fn duration_ms(&self) -> f64 {
+        self.decoder.duration_ms()
+    }
+
+    pub fn width(&self) -> u32 {
+        self.decoder.width()
+    }
+
+    pub fn height(&self) -> u32 {
+        self.decoder.height()
+    }
+
+    /// Returns a texture with the latest frame at the given timestamp.
+    ///
+    /// If the timestamp is negative, a zeroed texture is returned.
+    pub fn get_frame(&mut self, timestamp_s: f64) -> GpuTexture2D {
+        re_tracing::profile_function!();
+        self.decoder.get_frame(TimeMs::new(timestamp_s * 1e3))
```

Comment:
https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

```suggestion
    pub fn frame_at(&mut self, timestamp_s: f64) -> GpuTexture2D {
        re_tracing::profile_function!();
        self.decoder.frame_at(TimeMs::new(timestamp_s * 1e3))
```